R/theme.r

```diff
@@ -385,8 +364,30 @@ print.theme <- function(x, ...) utils::str(x)
 #' p + mytheme
 #'
 #' }
-theme <- function(..., complete = FALSE, validate = TRUE) {
-  elements <- list(...)
+theme <- function(line, rect, text, title, aspect.ratio, axis.title,
+                  axis.title.x, axis.title.x.top, axis.title.y,
+                  axis.title.y.right, axis.text, axis.text.x, axis.text.x.top,
+                  axis.text.y, axis.text.y.right, axis.ticks, axis.ticks.x,
+                  axis.ticks.y, axis.ticks.length, axis.line, axis.line.x,
+                  axis.line.y, legend.background, legend.margin, legend.spacing,
+                  legend.spacing.x, legend.spacing.y, legend.key,
+                  legend.key.size, legend.key.height, legend.key.width,
+                  legend.text, legend.text.align, legend.title,
+                  legend.title.align, legend.position, legend.direction,
+                  legend.justification, legend.box, legend.box.just,
+                  legend.box.margin, legend.box.background, legend.box.spacing,
+                  panel.background, panel.border, panel.spacing,
+                  panel.spacing.x, panel.spacing.y, panel.grid,
+                  panel.grid.major, panel.grid.minor, panel.grid.major.x,
+                  panel.grid.major.y, panel.grid.minor.x, panel.grid.minor.y,
+                  panel.ontop, plot.background, plot.title, plot.subtitle,
+                  plot.caption, plot.margin, strip.background, strip.text,
+                  strip.text.x, strip.text.y, strip.switch.pad.grid,
+                  strip.switch.pad.wrap, ..., complete = FALSE, validate = TRUE) {
+  elements <- c(as.list(environment()), list(...))
```

Comment:
Hmmmm, I don't like this but I don't see a better way except creating the massive named list by hand (which seems likely to be error prone).